### DISCUSSION TITLE
Fail to fetch stats error on performance dump?

### DISCUSSION THREAD

**Discussion by Ace7866** [mattdodge/yahoofantasy#14] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/58792096/120859058-4a041c80-c549-11eb-8a00-74dc41546c4b.png)

Hi guys! I'm so glad I was able to get everything working. Have you ever encountered this error when running the performance dump? Obviously this seems like an issue with Yahoo's database, and not the code, but I'm just curious. Looks like it makes about 3 attempts before moving on. As I type this the system is working on its third attempt for Peyton Barber but I'm curious to see how many more errors like this it encounters before finish (and how long in total the dump will take).

**Comment by Ace7866** [comment, marked as answer]
Ok. Dump just finished. Total time roughly 75-90 minutes. The bulk of this was probably waiting for the errors to get sorted (from screenshot above, ~6 minutes per error. Errors with the following: 

- Mason Rudolph   (390.p.31046), Week 3
- Peyton Barber      (390.p.29560), Week 5
- Vance McDonald (390.p.26678), Week 6
- Jimmy Graham    (390.p.24070), Week 8
- Darren Waller      (390.p.28592), Week 9
- Philip Rivers         (390.p.6763),   Week 11
- Devlin Hodges     (390.p.32581), Week 13
- Austin Ekeler        (390.p.30423), Week 14
- Mike Boone          (390.p.31228), Week 16

This is such a silly thing, but I can't seem to figure out if/where the CSV was actually created as a file on my system. Hopefully I can figure that out haha. I do see the entire output in my Command Prompt though!

**Reply by mattdodge** [reply]
I think that sounds a bit consistent with what I've seen too. The issue is due to Yahoo's rate limiting and how many requests it has to make to get all of the performance data. I think I briefly tried to combine the API requests to cut down on the number of them but it wasn't trivial to do so, at least that I could tell.

The good news is the data should be cached on your machine, so if you run it again it should go much faster.

If you don't specify any output (the `-o` flag in your command) the default is to just print to screen. Did it print to stdout or no? If not, try again and add `-o output.csv` to write to that file

**Reply by Ace7866** [reply]
Oh, that's good news re: the cache! 
Yes, it printed to stdout since I did not specify otherwise. I wound up copying the stdout and pasting directly into excel and then using the Text-to-Column function from the Data banner with the comma delimiter to split everything up into columns and then saved as a .csv. That got the job done! Next time I'll just add the command directly. 
This is so great! Now I have the massive .csv with all of the data and I'll be using it over the next couple days to practice Pandas (I just learned Pandas this past week). Hoping to parse, extract, and plot lots of interesting data from here. This data table will be great for putting my learning into practice.

**Comment by Ace7866** [comment]
![image](https://user-images.githubusercontent.com/58792096/122651091-3a193a80-d0fc-11eb-91c8-4c858f36bd3e.png)

hi @mattdodge, curious if you might've seen this error on the performances dump before? Today I decided I'd try pulling prior years using the dump command. Wondering how to troubleshoot or get around this one. Curiously it looks like it encountered its first error around the same time as my original screenshot above (19% and ~09:15 remaining). This leads me to think it might've been a failed fetching stats error, but rather than bypassing it as done above, it force quit the program and gave me this output instead.

**Reply by mattdodge** [reply]
Not sure I've seen this particular error before. Is there more to that stack trace? Looks like it might be cut off.

You're seeing the warning about parsing the XML response but I think that might just be what is expected when we're rate limited. The script should wait and try again later, that second request is maybe the one that is failing?
